pkg/controllers/management/node/cleanup.go

```diff
@@ -185,64 +198,77 @@ func (m *Lifecycle) waitUntilJobCompletes(userContext *config.UserContext, job *
 		return true, nil
-	}); err != nil && err.Error() != "timed out waiting for the condition" {
-		return err
-	}
+	})
+}
+
+func (m *Lifecycle) waitUntilJobCompletes(userContext *config.UserContext, job *v1.Job) error {
+	return m.waitForJobCondition(
+		userContext,
+		job,
+		func(j *v1.Job, err error) bool { return err == nil && j.Status.Succeeded > 0 },
+		"complete",
+	)
+}
 
-	// remove the job to clean up
-	return userContext.K8sClient.BatchV1().Jobs(job.Namespace).Delete(m.ctx, job.Name, metav1.DeleteOptions{PropagationPolicy: &[]metav1.DeletionPropagation{metav1.DeletePropagationForeground}[0]})
+func (m *Lifecycle) waitUntilJobDeletes(userContext *config.UserContext, job *v1.Job) error {
+	return m.waitForJobCondition(userContext, job, func(j *v1.Job, err error) bool {
+		if j.DeletionTimestamp.IsZero() {
+			err = userContext.BatchV1.Jobs(j.Namespace).Delete(j.Name, &metav1.DeleteOptions{PropagationPolicy: &[]metav1.DeletionPropagation{metav1.DeletePropagationForeground}[0]})
+		}
+		return kerror.IsNotFound(err)
+	},
+		"delete")
 }
 
-func (m *Lifecycle) createCleanupJob(userContext *config.UserContext, node *v3.Node) (*batchV1.Job, error) {
+func (m *Lifecycle) createCleanupJob(userContext *config.UserContext, cluster *v3.Cluster, node *v3.Node) (*batchV1.Job, error) {
 	nodeLabel := "cattle.io/node"
 
 	// find if someone else already kicked this job off
-	existingJob, err := userContext.K8sClient.BatchV1().Jobs("default").List(m.ctx, metav1.ListOptions{
+	existingJob, err := userContext.K8sClient.BatchV1().Jobs(namespace.System).List(m.ctx, metav1.ListOptions{
 		LabelSelector: fmt.Sprintf("%s=%s", nodeLabel, node.Name),
 	})
 	if err != nil && !kerror.IsNotFound(err) {
 		if strings.Contains(err.Error(), dialer.ErrAgentDisconnected.Error()) ||
 			strings.Contains(err.Error(), "connection refused") {
-			return nil, nil // can't connect, just continue on with the delete
+			return nil, nil // can't connect, just continue on with deleting v3 node
 		}
 		return nil, err
 	}
 
 	if len(existingJob.Items) != 0 {
-		logrus.Debugf("found existing jobs for node: %s", node.Name)
-		// found existing job by label selector so fake an already exists
-		return nil, &kerror.StatusError{
-			ErrStatus: metav1.Status{Reason: metav1.StatusReasonAlreadyExists},
+		if existingJob.Items[0].DeletionTimestamp.IsZero() {
+			// found an existing job that isn't deleting, so assuming another run of the controller is working on it.
+			// Return an "already exists" error.
+			return nil, &kerror.StatusError{
+				ErrStatus: metav1.Status{
+					Reason:  metav1.StatusReasonAlreadyExists,
+					Message: fmt.Sprintf("job already exists for %s/%s", node.Namespace, node.Name),
+				},
+			}
 		}
+
+		return nil, m.waitUntilJobDeletes(userContext, &existingJob.Items[0])
 	}
 
 	meta := metav1.ObjectMeta{
-		GenerateName: "cattle-node-cleanup-",
-		Namespace:    "default",
+		GenerateName: "cattle-cleanup-node-",
+		Namespace:    namespace.System,
```

Comment:
i think we should leave this in default for "reasons". there are more cleanup jobs in default that would need to be changed if you stick with this choice.